src/microservices/payment/graphql/payment-mutation.resolver.ts

```diff
@@ -0,0 +1,39 @@
+import { Args, Mutation, Resolver } from '@nestjs/graphql';
+import { GqlPayment } from './models/payment-gql.model';
+import { CreatePaymentInput } from './input/create-payment.input';
+import { UseGuards } from '@nestjs/common';
+import { GqlJwtAuthGuard } from '../../auth/graphql/guards';
+import { PaymentService } from '../payment.service';
+
+@Resolver((of) => GqlPayment)
+export class PaymentMutationResolver {
+  constructor(private readonly paymentService: PaymentService) {}
+
+  @UseGuards(GqlJwtAuthGuard)
+  @Mutation((returns) => GqlPayment)
+  async createPayment(
+    @Args('createPaymentData')
+    createPaymentInput: CreatePaymentInput,
+  ): Promise<GqlPayment> {
+    const newPayment = await this.paymentService.createPayment(
```

Comment:
As this just returns a value from a service it can be done like that
```suggestion
    return this.paymentService.createPayment(createPaymentInput);
```

It may look strange that we have async method and no await, but we have await in payment service.